src/modules/75-cf/components/GitSyncSetupModal/GitSyncSetupModal.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useCallback, useRef } from 'react'
+import { useParams } from 'react-router-dom'
+import { FontVariation, Color } from '@harness/design-system'
+import type { FormikProps } from 'formik'
+import * as Yup from 'yup'
+import { Button, ButtonVariation, Formik, FormikForm, Heading, Layout, ModalDialog, useToaster } from '@harness/uicore'
+import { fullYamlPathRegex } from '@common/utils/StringUtils'
+import { useStrings } from 'framework/strings'
+import { GitSyncForm, GitSyncFormFields } from '@gitsync/components/GitSyncForm/GitSyncForm'
+import { useCreateGitRepo } from 'services/cf'
+import { getErrorMessage } from '@cf/utils/CFUtils'
+
+interface GitSyncSetupModalProps {
+  isOpen: boolean
+  hideModal: () => void
+}
+
+export const GitSyncSetupModal: React.FC<GitSyncSetupModalProps> = ({ isOpen, hideModal }) => {
+  const { orgIdentifier, projectIdentifier, accountId: accountIdentifier } = useParams<Record<string, string>>()
+  const { showError } = useToaster()
+  const { getString } = useStrings()
+
+  const formikRef = useRef<FormikProps<GitSyncFormFields>>()
+
+  const { mutate: createGitRepo } = useCreateGitRepo({
+    identifier: projectIdentifier,
+    queryParams: {
+      accountIdentifier,
+      orgIdentifier
+    }
+  })
+
+  const handleSubmit = useCallback(
+    async (formValues: GitSyncFormFields) => {
+      // get file name and root folder path from filePath field
+      const fullPath = formValues?.filePath
+      const pathNodes = fullPath?.split('/').filter((node: string) => !!node) // remove empty
+      const yamlFileName = '/' + pathNodes?.pop()
+      const yamlFilePath = pathNodes?.length ? '/' + pathNodes?.join('/') + '/' : ''
+
+      const requestData = {
+        repoIdentifier: formValues.repo,
+        repoReference: formValues.repo,
+        branch: formValues.branch,
+        filePath: yamlFileName,
+        rootFolder: yamlFilePath,
+        connectorReference:
+          typeof formValues.connectorRef === 'string' ? formValues.connectorRef : formValues.connectorRef?.value
+      }
+
+      try {
+        await createGitRepo(requestData)
+
+        hideModal()
+      } catch (error) {
+        showError(getErrorMessage(error), 0, getString('cf.selectFlagRepo.createRepoError'))
+      }
+    },
+    [createGitRepo, getString, hideModal, showError]
+  )
+
+  return (
+    <ModalDialog
+      isOpen={isOpen}
+      enforceFocus={false}
+      title={
+        <Heading level={4} font={{ variation: FontVariation.H4 }} color={Color.GREY_800}>
+          {getString('cf.gitSync.setUpGitConnection')}
+        </Heading>
+      }
```

Comment:
The title already sets as a heading, you just need to pass the text
```suggestion
      title={getString('cf.gitSync.setUpGitConnection')}
```